`NULLIF` panics when `expr` is null

```sql
SELECT NULLIF(null, 0);
```

### Expected
| col1 |
| -- |
| null |

### Actual
```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x100e7fae2]

goroutine 14 [running]:
panic({0x1021da500?, 0x103a81230?})
	/usr/local/Cellar/go/1.21.6/libexec/src/runtime/panic.go:914 +0x21f
github.com/goccy/go-zetasqlite/internal.NULLIF({0x0, 0x0}, {0x1026acd80?, 0x10389fac0?})
```

--- a/internal/function.go
+++ b/internal/function.go
@@ -384,6 +384,9 @@ func IFNULL(expr, nullResult Value) (Value, error) {
 }
 
 func NULLIF(expr, exprToMatch Value) (Value, error) {
+	if expr == nil {
+		return nil, nil
+	}
 	cond, err := expr.EQ(exprToMatch)
 	if err != nil {
 		return nil, err

--- a/query_test.go
+++ b/query_test.go
@@ -471,6 +471,11 @@ FROM UNNEST([1, 2, 3, 4]) AS val`,
 			expectedRows: [][]interface{}{{int64(10)}},
 		},
 		{
+			name:         "nullif null",
+			query:        `SELECT NULLIF(null, 0)`,
+			expectedRows: [][]interface{}{{nil}},
+		},
+		{
 			name:         "rounding",
 			query:        `SELECT ROUND(2.0), ROUND(2.3), ROUND(2.8), ROUND(2.5), ROUND(-2.3), ROUND(-2.8), ROUND(-2.5)`,
 			expectedRows: [][]interface{}{{float64(2.0), float64(2.0), float64(3.0), float64(3.0), float64(-2.0), float64(-3.0), float64(-3.0)}},

